rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala

```diff
@@ -41,6 +43,38 @@ class Runtime private (
     replaySpace.close()
     context.close()
   }
+  def injectEmptyRegistryRoot[F[_]]()(implicit F: Sync[F]): F[Unit] = {
```

Comment:
Empty parentheses are only needed to indicate a method with no arguments has side effects. I think you can drop the parentheses and wrap the `produce` calls in `Sync[F].delay` in order to leave this as a pure method.
```scala
for {
  spaceResult <- F.delay(space.produce(...))
  replayResult <- F.delay(replaySpace.produce(...))
  _ <- spaceResult match { ... }
} yield ()
```

Reply:
Isn't this going to duplicate the work done in #1586 @birchmd ?

Reply:
I'll let your PR go though first and then update mine to call `injectEmptyRegistryRoot`

Reply:
I've wrapped the side effects in F.